### PR TITLE
LibWeb: Implement HTMLVideoElement `load`, `play`, `pause` and draw some basic controls

### PR DESCRIPTION
**Pull request by trflynn89**
https://user-images.githubusercontent.com/5600524/230734615-8d7a62b9-a8c0-4eec-a9e5-1efe4177a644.mp4

### CODE REVIEW

**Comment by trflynn89** [comment]
Edit: Realized the playback buttons should have some feedback on hover, so I added that to the layout node

File: Userland/Libraries/LibWeb/Painting/VideoPaintable.cpp [hunk 1]
```diff
@@ -40,11 +60,153 @@ void VideoPaintable::paint(PaintContext& context, PaintPhase phase) const
     if (phase != PaintPhase::Foreground)
         return;
 
+    auto video_rect = context.rounded_device_rect(absolute_rect());
+    ScopedCornerRadiusClip corner_clip { context, context.painter(), video_rect, normalized_border_radii_data(ShrinkRadiiForBorders::Yes) };
+
+    Optional<Web::DevicePixelPoint> mouse_position;
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Painting/VideoPaintable.cpp:66]: Omit Web namespace

File: Userland/Libraries/LibWeb/Painting/VideoPaintable.cpp [hunk 2]
```diff
@@ -40,11 +60,153 @@ void VideoPaintable::paint(PaintContext& context, PaintPhase phase) const
...
+    } else {
+        auto input_colors = compute_input_colors(context.palette(), computed_values().accent_color());
+        context.painter().fill_rect(video_rect.to_type<int>(), input_colors.light_gray);
+
+        if (paint_user_agent_controls)
+            paint_placeholder_video_controls(context, video_rect, mouse_position);
+    }
+}
+
+void VideoPaintable::paint_loaded_video_controls(PaintContext& context, HTML::HTMLVideoElement const& video_element, Web::DevicePixelRect video_rect, Optional<Web::DevicePixelPoint> const& mouse_position) const
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Painting/VideoPaintable.cpp:87]: Likewise

File: Userland/Libraries/LibWeb/Painting/VideoPaintable.h [hunk 3]
```diff
@@ -19,10 +20,20 @@ class VideoPaintable final : public PaintableBox {
...
     Layout::VideoBox const& layout_box() const;
 
 private:
     VideoPaintable(Layout::VideoBox const&);
+
+    virtual bool wants_mouse_events() const override { return true; }
+    virtual DispatchEventOfSameName handle_mouseup(Badge<EventHandler>, CSSPixelPoint, unsigned button, unsigned modifiers) override;
+    virtual DispatchEventOfSameName handle_mousemove(Badge<EventHandler>, CSSPixelPoint, unsigned buttons, unsigned modifiers) override;
+
+    void paint_loaded_video_controls(PaintContext&, HTML::HTMLVideoElement const&, Web::DevicePixelRect video_rect, Optional<Web::DevicePixelPoint> const& mouse_position) const;
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Painting/VideoPaintable.h:33]: Likewise

**Review by linusg — CHANGES_REQUESTED** [review]

**Pull request merged**